Common/GPU/Vulkan/VulkanContext.cpp

```diff
@@ -1537,3 +1549,65 @@ std::string FormatDriverVersion(const VkPhysicalDeviceProperties &props) {
 	uint32_t branch = VK_VERSION_PATCH(props.driverVersion);
 	return StringFromFormat("%d.%d.%d (%08x)", major, minor, branch, props.driverVersion);
 }
+
+// Mainly just the formats seen on gpuinfo.org for swapchains, as this function is only used for listing
+// those in the UI. Also depth buffers that we used in one place.
+// Might add more in the future if we find more uses for this.
+const char *VulkanFormatToString(VkFormat format) {
+	switch (format) {
+	case VK_FORMAT_A1R5G5B5_UNORM_PACK16: return "A1R5G5B5_UNORM_PACK16";
+	case VK_FORMAT_A2B10G10R10_UNORM_PACK32: return "A2B10G10R10_UNORM_PACK32";
+	case VK_FORMAT_A8B8G8R8_SNORM_PACK32: return "A8B8G8R8_SNORM_PACK32";
+	case VK_FORMAT_A8B8G8R8_SRGB_PACK32: return "A8B8G8R8_SRGB_PACK32";
+	case VK_FORMAT_A8B8G8R8_UNORM_PACK32: return "A8B8G8R8_UNORM_PACK32";
+	case VK_FORMAT_B10G11R11_UFLOAT_PACK32: return "B10G11R11_UFLOAT_PACK32";
+	case VK_FORMAT_B4G4R4A4_UNORM_PACK16: return "B4G4R4A4_UNORM_PACK16";
+	case VK_FORMAT_B5G5R5A1_UNORM_PACK16: return "B5G5R5A1_UNORM_PACK16";
+	case VK_FORMAT_B5G6R5_UNORM_PACK16: return "B5G6R5_UNORM_PACK16";
+	case VK_FORMAT_B8G8R8A8_SNORM: return "B8G8R8A8_SNORM";
+	case VK_FORMAT_B8G8R8A8_SRGB: return "B8G8R8A8_SRGB";
+	case VK_FORMAT_B8G8R8A8_UNORM: return "B8G8R8A8_UNORM";
+	case VK_FORMAT_R16G16B16A16_SFLOAT: return "R16G16B16A16_SFLOAT";
+	case VK_FORMAT_R16G16B16A16_SNORM: return "R16G16B16A16_SNORM";
+	case VK_FORMAT_R16G16B16A16_UNORM: return "R16G16B16A16_UNORM";
+	case VK_FORMAT_R4G4B4A4_UNORM_PACK16: return "R4G4B4A4_UNORM_PACK16";
+	case VK_FORMAT_R5G5B5A1_UNORM_PACK16: return "R5G5B5A1_UNORM_PACK16";
+	case VK_FORMAT_R5G6B5_UNORM_PACK16: return "R5G6B5_UNORM_PACK16";
+	case VK_FORMAT_R8G8B8A8_SNORM: return "R8G8B8A8_SNORM";
+	case VK_FORMAT_R8G8B8A8_SRGB: return "R8G8B8A8_SRGB";
+	case VK_FORMAT_R8G8B8A8_UNORM: return "R8G8B8A8_UNORM";
+
+	case VK_FORMAT_D24_UNORM_S8_UINT: return "D24S8";
+	case VK_FORMAT_D16_UNORM: return "D16";
+	case VK_FORMAT_D16_UNORM_S8_UINT: return "D16S8";
+	case VK_FORMAT_D32_SFLOAT: return "D32f";
+	case VK_FORMAT_D32_SFLOAT_S8_UINT: return "D32fS8";
+	case VK_FORMAT_S8_UINT: return "S8";
+	case VK_FORMAT_UNDEFINED: return "UNDEFINED (BAD!)";
+
+	default: return "(format not added to string list)";
+	}
+}
+
+// I miss Rust where this is automatic :(
```

Comment:
Apparently it's not even happening until C++23 now, at best...

-[Unknown]

Reply:
Another thing I miss from Rust is the robust namespacing. The kind of name collision that caused the build failure is a complete non-issue there.